staging/src/k8s.io/client-go/go.mod

```diff
@@ -44,13 +47,17 @@ require (
 	github.com/josharian/intern v1.0.0 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
 	github.com/mailru/easyjson v0.7.7 // indirect
+	github.com/matttproud/golang_protobuf_extensions v1.0.4 // indirect
 	github.com/moby/spdystream v0.2.0 // indirect
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.2 // indirect
 	github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822 // indirect
 	github.com/onsi/gomega v1.27.6 // indirect
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
+	github.com/prometheus/client_model v0.3.0 // indirect
+	github.com/prometheus/common v0.42.0 // indirect
+	github.com/prometheus/procfs v0.9.0 // indirect
```

Comment:
The entire point of the client-go abstraction was to get rid of the prometheus dependency, we cannot do this. You can already inject prometheus using the interface that client-go describes.

Reply:
> You can already inject prometheus using the interface that client-go describes.

So is your recommendation that users write their own Prometheus adapter code, or should they use `k8s.io/component-base/metrics/prometheus/clientgo` as suggested in another comment?